### ISSUE TITLE
virt-whitespace file looks wrong on disk

### ISSUE THREAD

**Issue by eisbehr** [4coder-editor/4coder#133]
When a virtual whitespace file is written out on disk, the resulting file has indentation errors that are not consistent with anything. Not with how it looks in 4coder, and not with any reasonable indentation.
example: 
```C
    u32 parse_args(ctx *c, op_mnemonic Mnemonic, b32 HasRd, b32 HasRs1, b32 HasRs2, b32 HasImm) {
    u32 Rd = 0;
    u32 Rs1 = 0;
    u32 Rs2 = 0;
    u32 Imm = 0;
        if(HasRd) Rd = parse_reg(c);
                   if(HasRs1) Rs1 = parse_reg(c);
        if(HasRs2) Rs2 = parse_reg(c);
    if(HasImm) Rs2 = parse_imm(c);
    return op(Mnemonic, Rd, Rs1, Rs2, Imm);
}
```

**Comment by exanimusfan** [comment]
I would argue this isn't a bug. You might not understand what virtual whitespace is. In order to fix the way identation looks you need to turn virtual whitespace off and fix it visualy. There is also a option in your config file "automatically_indent_text_on_save = true;" that way it auto-indents the text before saving it to disk.

**Comment by eisbehr** [comment]
You are free to argue that, but you are wrong. 
This is set in my config file `automatically_indent_text_on_save = true;` and the alpha used to indent correctly. I've used 4coder for years now, this behavior is new. And obviously buggy behavior.

**Comment by tebtro** [comment]
So in `BUFFER_HOOK_SIG(default_file_save)` it is checking for automatically_indent_text_on_save,
but it is also checking if you are in virtual whitespace mode and that variable is just not set.

```
BUFFER_HOOK_SIG(default_file_save){
    // buffer_id
    ProfileScope(app, "default file save");
    b32 is_virtual = false;
    if (global_config.automatically_indent_text_on_save && is_virtual){
        auto_indent_buffer(app, buffer_id, buffer_range(app, buffer_id));
    }
```

**Comment by eisbehr** [comment]
Great find, that seems to be the problem!

**Comment by eisbehr** [comment]
Actually, that still seems broken. Maybe Allen left it off because of that.

**Comment by 4coder-editor** [comment]
Thanks for finding the bug and the fix. The fix is in 4.1.1.
